main.c

```diff
@@ -96,10 +111,42 @@ static void process_share_deprecated(struct entry_settings *opts, const char *op
 	}
 }
 
-static void handle_limit_arg(int resource, struct rlimit *opt_limits, char *arg, char const *which ) {
-	if (parse_rlimit(resource, opt_limits + resource, arg)) {
-		err(1, "error in --limit-%s value", which);
+static void handle_limit_arg(int option_num, struct entry_settings *opts, char *arg) {
+	const struct {
```

Comment:
Since the map isn't mutated, I'd make it static. I almost missed the *opt_ent, so I don't mind if these variable declarations were separate, I find it clearer to read. You'd have to name your struct, but eh, it isn't too bad given that it can be defined in the function itself.

Reply:
> Looks very nice, thank you. I have a few comments, mostly nits.
> 
> For some reason, on alpine RLIMIT_RTTIME isn't defined in sys/resource.h, but get defined by asm/resource.h. I don't mind importing asm/resource.h in main.c if that does not cause any other conflicts.

:facepalm: I saw the compilation error and thought it was due to the -Wconversion errors. haha.  Anyway, those are fixed, now :dancers:

Reply:
> Since the map isn't mutated, I'd make it static.

TIL.  I thought that `const struct foo[] = {...};` would just plunk it in `.rodata` and be done with it!

Reply:
It only goes in rodata if it's `static const` (at least, if it's a function variable; for top-level variables the const is sufficient)